mindsdb/interfaces/agents/langchain_agent.py

```diff
@@ -439,33 +457,52 @@ def _invoke_agent_executor_with_prompt(agent_executor, prompt):
 
-        Returns:
-            chunks (Iterable[Dict]): Completion chunks for agent
-        '''
-        # Prefer prediction time prompt template, if available.
-        base_template = args.get('prompt_template', args['prompt_template'])
-        input_variables = []
-        matches = list(re.finditer("{{(.*?)}}", base_template))
-
-        for m in matches:
-            input_variables.append(m[0].replace('{', '').replace('}', ''))
-
-        base_template = base_template.replace('{{', '{').replace('}}', '}')
-        first_row = df.iloc[0]
-        prompt_template = PromptTemplate(input_variables=input_variables, template=base_template)
-        kwargs = {}
-        for col in input_variables:
-            kwargs[col] = first_row[col] if first_row[col] is not None else ''  # add empty quote if data is missing
-        prompt = prompt_template.format(**kwargs)
-        if not prompt:
-            return
-        for chunk in agent_executor.stream(prompt, config={'callbacks': self._get_agent_callbacks(args)}):
-            yield chunk
+        callbacks, context_callback = prepare_callbacks(self, args)
+
+        yield 'data: {}\n\n'.format(json.dumps({"type": "start", "prompt": prompts[0]}))
+
+        try:
+            # Check if stream method exists
+            if not hasattr(agent_executor, 'stream') or not callable(agent_executor.stream):
+                raise AttributeError("The agent_executor does not have a 'stream' method")
+
+            # Try to get the stream iterator
+            stream_iterator = agent_executor.stream(prompts[0], config={'callbacks': callbacks})
+
+            # Ensure it's iterable
+            if not hasattr(stream_iterator, '__iter__'):
+                raise TypeError("The stream method did not return an iterable")
+
+            for chunk in stream_iterator:
+                chunk_obj = process_chunk(chunk)
+                yield 'data: {}\n\n'.format(json.dumps(chunk_obj))
+
+            # After the stream is complete, yield the captured context
+            captured_context = context_callback.get_contexts()
+            if captured_context:
+                yield 'data: {}\n\n'.format(json.dumps({
+                    "type": "context",
+                    "content": captured_context
+                }))
+
+        except AttributeError as e:
+            error_message = f"Stream method error: {str(e)}"
+            logger.error(error_message, exc_info=True)
+            yield 'data: {}\n\n'.format(json.dumps({"error": error_message}))
+
+        except TypeError as e:
+            error_message = f"Stream iterator error: {str(e)}"
+            logger.error(error_message, exc_info=True)
+            yield 'data: {}\n\n'.format(json.dumps({"error": error_message}))
+
+        except Exception as e:
+            error_message = handle_agent_error(e)
+            yield 'data: {}\n\n'.format(json.dumps({"error": error_message}))
+
+        finally:
```

Comment:
if this message has to be sent: maybe it is safer to move it to _completion_event_generator or to route function.
potentially the stream_agent could not be called